src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
         Assert.Equal(value.IntProperty, actual.IntProperty);
     }
 
+    #region DesktopTest
+
+    /// <summary>
+    /// Roundtrips a Datacontract type  which contains Primitive types assigned to member of type object. 
+    /// Resolver is plugged in and resolves the primitive types. Verify resolver called during ser and deser
+    /// </summary>
+    [Fact]
+    public static void DCS_BasicRoundTripResolvePrimitiveTypes()
+    {
+        var dataContractSerializerSettings = new DataContractSerializerSettings()
+        {
+            DataContractResolver = new DesktopTestData.PrimitiveTypeResolver()
+            , IgnoreExtensionDataObject = false
+            , KnownTypes = null
+            , MaxItemsInObjectGraph = int.MaxValue
+            , PreserveObjectReferences = false
+        };
+
+        string baseline = ""; 
```

Comment:
The object has some fields which are using **dynamic generated value**( such as DateTime.Now and Guid.NewGuid() ),I will try to use hard coded value instead and set expected value to `baseline`.